core/frontend/src/components/version-chooser/VersionChooser.vue

```diff
@@ -234,25 +234,77 @@ export default Vue.extend({
     this.loadCurrentVersion()
   },
   methods: {
-    async checkIfBackendIsOnline() {
-      this.waiting = true
-      await back_axios({
-        method: 'get',
-        url: '/version-chooser/v1.0/version/current',
-        timeout: 500,
-      })
-        .then(() => {
-          if (this.waiting) {
-            // Allow 3 seconds so the user can read the "complete" message
-            // reload(true) forces the browser to fetch the page again
-            setTimeout(() => { window.location.reload(true) }, 3000)
-          }
-        })
-        .catch((error) => {
-          console.log(error)
-          this.waiting = true
+    backendIsOnline() {
+      return new Promise((resolve) => {
+        back_axios({
+          method: 'get',
+          url: '/version-chooser/v1.0/version/current',
+          timeout: 500,
         })
+          .then(() => {
+            resolve(true)
+          })
+          .catch(() => {
+            resolve(false)
+          })
+      })
     },
+
+    async waitForBackendToGoOffline() {
+      let timeout = 0
+      let interval = 0
+      return new Promise((resolve, reject) => {
+        timeout = setTimeout(
+          () => {
+            reject(new Error('backend took to long to shutdown!'))
+            clearInterval(interval)
+          },
+          20000,
+        )
+        interval = setInterval(() => {
+          this.backendIsOnline().then((backend_online) => {
+            if (!backend_online) {
+              clearTimeout(timeout)
+              clearInterval(interval)
+              resolve('backend went offline')
+            }
+          })
+        }, 1000)
+      })
+    },
+
+    waitForBackendToGoOnline() {
+      let timeout = 0
+      let interval = 0
+      return new Promise((resolve, reject) => {
+        timeout = setTimeout(
+          () => {
+            reject(new Error('backend took to long to come back'))
+            clearInterval(interval)
+          },
+          20000,
+        )
+        interval = setInterval(() => {
+          this.backendIsOnline().then((backend_online) => {
+            if (backend_online) {
+              clearTimeout(timeout)
+              clearInterval(interval)
+              resolve('backend went online')
+            }
+          })
+        }, 1000)
+      })
+    },
+
+    async waitForBackendToRestart(reload: boolean) {
+      this.waiting = true
```

Comment:
Just to be sure, there is no need to set waiting to false since the logic here is to reload the page, right ?

Reply:
setting waiting to true shows up the overlay and spinner.